src/test/kotlin/three/consulting/epoc/repository/ProjectServiceIntegrationTest.kt

```diff
@@ -38,13 +42,15 @@ class ProjectServiceIntegrationTest : IntegrationTest() {
         val project = ProjectDTO(
             name = "Sample",
             description = "Sample project",
-            startingDate = LocalDate.now(),
-            endDate = LocalDate.now()
+            startDate = LocalDate.now(),
+            endDate = LocalDate.now(),
+            customer = customerService.findCustomerForId(1L)!!,
+            managingEmployee = employeeService.findEmployeeForId(1L)!!
```

Comment:
You shouldn't bring the other services in for this. In tests the test values should be hard coded and not fetched from the database. So you can just create new customer and managingEmployee here without calling them from the database.

Reply:
So creating new values like this:
```
customer = customerService.createCustomer(CustomerDTO(null, "Project Test customer", "Innovating new innovation!", enabled = true)),
managingEmployee = employeeService.createEmployee(EmployeeDTO(null, "project", "test", "test@project.entity")),  
```
As far as I can see the objects used in the relation have to be in the DB so I have to either create them or fetch them?

Reply:
Actually it seems that the following works:
```
customer = CustomerDTO(1, "Maurin Makkara Oy"),
managingEmployee = EmployeeDTO(1, "Testi", "Tekijä", "testi@tekija.fi"),
```
As long as the values match the ones that are in the db. So I will remove the other services.

Reply:
Nope use the constructor directly. So 
```kotlin
customer = Customer(1L, "Project Test customer", "Innovating new innovation!")
```
etc with an id that already has an existing customer in the database. So to not create new one.

Reply:
The usecase for creating a project would be to use a pre-existing related objects as customer and managing employee, so this test should do just that.

Reply:
Yes makes total sense. I just didn't realise it was possible so didn't even try! Thanks again for the tips!